### ISSUE TITLE
'pre-binding' dynamic columns?

### ISSUE THREAD

**Issue by matthew-wozniczka** [Microsoft/ODBC-Specification#102]
If the application binds column `N + 1` (with the cursor currently having `N` columns), and a fetch is about to be interrupted because a new column was found (Let's assume `SQL_ATTR_TYPE_EXCEPTION_BEHAVIOR` is _not_ `SQL_TE_REPORT_ALL`, I assume that would apply here), should the driver simply use that binding instead of returning `SQL_DATA_AVAILABLE`?

**Comment by mikepizzo** [comment]
Yes; the driver should use whatever bindings are available at the time it is performing the fetch.   We do say that the application cannot change binding information for columns other than the "current" column, but if the application binds more columns than are described, those bindings should be used at fetch.

**Comment by mikepizzo** [comment]
Check: does the Driver manager allow binding more columns than currently described by the descriptor?

**Comment by mikepizzo** [comment]
Spec calls out SQLState 07009, Invalid descriptor index, with the condition "The value specified for the argument ColumnNumber exceeded the maximum number of columns in the result set."

Not sure if DM enforces this currently or not.
